src/hooks/useFetch.ts

```diff
@@ -41,55 +43,91 @@ export function useFetch<T>(options: UseFetchOptions) {
     skip = false,
     auto = true,
   } = options;
+
   const [data, setData] = useState<T | null>(null);
+
   const [loading, setLoading] = useState(false);
+
   const [error, setError] = useState<Error | null>(null);
 
   const fetchData = useCallback(
     async (overrideOptions?: Partial<UseFetchOptions>) => {
       setLoading(true);
+
       setError(null);
+
       try {
         const url = getRequestUrl({
           baseUrl: overrideOptions?.baseUrl ?? baseUrl,
           path: overrideOptions?.path ?? path,
           searchParams: overrideOptions?.searchParams ?? searchParams,
         });
+
         const res = await fetch(url, {
           method: overrideOptions?.method ?? method,
+
           headers: overrideOptions?.headers ?? headers,
+
           body: overrideOptions?.body
             ? JSON.stringify(overrideOptions.body)
             : body
               ? JSON.stringify(body)
               : undefined,
         });
+
         if (!res.ok) {
-          throw new Error('@kakao.com 이메일 주소만 가능합니다.');
+          let errData: Record<string, unknown> = {};
+          try {
+            errData = await res.json();
+          } catch {}
+          const errorMsg =
+            typeof errData.message === 'string'
+              ? errData.message
+              : res.statusText;
+          const error = new Error(errorMsg);
+          (
+            error as Error & {
+              response?: Record<string, unknown>;
+              status?: number;
+            }
+          ).response = errData;
+          (
+            error as Error & {
+              response?: Record<string, unknown>;
+              status?: number;
+            }
+          ).status = res.status;
+          throw error;
```

Comment:
[변경 요청] fetch 에러 응답에 대한 커스텀 에러를 정의하고 해당 커스텀 에러에 대한 타입 가드를 구현하면 에러를 사용하는 곳에서 에러 추론을 명확히 할 수 있어요.

```suggestion
          throw new FetchError(response);
```

```ts
// 커스텀 에러 클래스
class FetchError extends Error {
  constructor(...) {}
}

// 타입가드
function isFetchError(err: unknown): err is FetchError {
  return err instanceof FetchError;
}

// 컴포넌트에서 사용 예시
if (isFetchError(error) && error.status === 404) {
  navigate('/', { replace: true });
}
```